projects/aca-content/src/lib/components/search/search-input.service.ts

```diff
@@ -23,17 +23,33 @@
  * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Component, Input } from '@angular/core';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
 
-/**
- * This wrapper is designated to be used with 'adf-dynamic-component'.
- * It forwards the dynamic component inputs to original sidenav.
- */
-@Component({
-  selector: 'aca-sidenav-wrapper',
-  templateUrl: './sidenav-wrapper.component.html'
+@Injectable({
+  providedIn: 'root'
 })
-export class SidenavWrapperComponent {
-  @Input()
-  data: { mode?: 'collapsed' | 'expanded' } = {};
+export class SearchInputService {
+  savedRoute = '';
+
+  constructor(private router: Router) {}
+
+  isSearchRoute(): boolean {
+    return this.router.url.includes('/search');
+  }
+
+  saveRoute(route: string) {
+    this.savedRoute = route;
+  }
+
+  exitSearch() {
+    if (this.savedRoute.length > 0) {
```

Comment:
What if someone will call `saveRoute` with `null`? I think you should add null check for `savedRoute` here